[BUG] OIIO 2.5.4 Build failure on Fedora 32bit

**Describe the bug**
I know 32 bit isn't really supported but was hoping it was a simple fix.

**To Reproduce**
Attempt to build for x86 32-bit

**Evidence**
```
In file included from /builddir/build/BUILD/OpenImageIO-2.5.4.0/src/libtexture/texturesys.cpp:33:
/builddir/build/BUILD/OpenImageIO-2.5.4.0/src/libtexture/imagecache_pvt.h: In member function ‘size_t OpenImageIO_v2_5::pvt::TileID::hash() const’:
/builddir/build/BUILD/OpenImageIO-2.5.4.0/src/libtexture/imagecache_pvt.h:571:46: error: static assertion failed: FastHash uses the fewest instructions when data size is a multiple of 8 bytes.
  571 |             sizeof(*this) % sizeof(uint64_t) == 0,
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
/builddir/build/BUILD/OpenImageIO-2.5.4.0/src/libtexture/imagecache_pvt.h:571:46: note: the comparison reduces to ‘(4 == 0)’
gmake[2]: *** [src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/build.make:513: src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/__/libtexture/texturesys.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
In file included from /builddir/build/BUILD/OpenImageIO-2.5.4.0/src/libtexture/texture3d.cpp:23:
/builddir/build/BUILD/OpenImageIO-2.5.4.0/src/libtexture/imagecache_pvt.h: In member function ‘size_t OpenImageIO_v2_5::pvt::TileID::hash() const’:
/builddir/build/BUILD/OpenImageIO-2.5.4.0/src/libtexture/imagecache_pvt.h:571:46: error: static assertion failed: FastHash uses the fewest instructions when data size is a multiple of 8 bytes.
  571 |             sizeof(*this) % sizeof(uint64_t) == 0,
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
/builddir/build/BUILD/OpenImageIO-2.5.4.0/src/libtexture/imagecache_pvt.h:571:46: note: the comparison reduces to ‘(4 == 0)’
gmake[2]: *** [src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/build.make:527: src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/__/libtexture/texture3d.cpp.o] Error 1
gmake[2]: Leaving directory '/builddir/build/BUILD/OpenImageIO-2.5.4.0/redhat-linux-build'
gmake[1]: Leaving directory '/builddir/build/BUILD/OpenImageIO-2.5.4.0/redhat-linux-build'
gmake[1]: *** [CMakeFiles/Makefile2:1704: src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/all] Error 2
gmake: *** [Makefile:169: all] Error 2
```

Full build log:
https://kojipkgs.fedoraproject.org//work/tasks/1900/107051900/build.log

**Platform information:**
 - OIIO branch/version: 2.5.4
 - OS: Fedora
 - C++ compiler: GCC 13
 - Any non-default build flags when you build OIIO: A lot :)

--- a/src/libtexture/imagecache_pvt.h
+++ b/src/libtexture/imagecache_pvt.h
@@ -567,9 +567,11 @@ struct TileID {
         static_assert(
             sizeof(*this) == member_size,
             "All TileID members must be accounted for so we can hash the entire class.");
+#ifdef __LP64__
         static_assert(
             sizeof(*this) % sizeof(uint64_t) == 0,
             "FastHash uses the fewest instructions when data size is a multiple of 8 bytes.");
+#endif
         return fasthash::fasthash64(this, sizeof(*this));
     }
 

